### PR TITLE
SC-663-api-add-contiuations-list

### PR DESCRIPTION
**Pull request by potan**

### CODE REVIEW

File: node/src/main/scala/com/wavesplatform/api/http/TransactionsApiRoute.scala [hunk 1]
```diff
@@ -75,12 +75,17 @@ case class TransactionsApiRoute(
   private def loadTransactionStatus(id: ByteStr): JsObject = {
     import Status._
     val statusJson = blockchain.transactionInfo(id) match {
-      case Some((height, _, succeeded)) =>
+      case Some((height, t, succeeded)) =>
+        val enrich = t match {
+          case t: smart.InvokeScriptTransaction => Json.obj("сontinuationTtansactionIds" -> commonApi.continuations(t.id()).map(_.toString))
```
Comment (xrtm000) [inline node/src/main/scala/com/wavesplatform/api/http/TransactionsApiRoute.scala:80]:
```suggestion
          case t: smart.InvokeScriptTransaction => Json.obj("сontinuationTransactionIds" -> commonApi.continuations(t.id()).map(_.toString))
```

File: node/src/main/scala/com/wavesplatform/api/http/TransactionsApiRoute.scala [hunk 2]
```diff
@@ -75,12 +75,17 @@ case class TransactionsApiRoute(
...
+          case t: smart.ContinuationTransaction => Json.obj("сontinuationTtansactionIds" -> commonApi.continuations(t.invokeScriptTransactionId).map(_.toString))
```
Comment (xrtm000) [inline node/src/main/scala/com/wavesplatform/api/http/TransactionsApiRoute.scala:81]:
```suggestion
          case t: smart.ContinuationTransaction => Json.obj("сontinuationTransactionIds" -> commonApi.continuations(t.invokeScriptTransactionId).map(_.toString))
```

**Review by xrtm000 — CHANGES_REQUESTED** [review]
Test in ContinuationSuite

File: node/src/main/scala/com/wavesplatform/api/http/TransactionsApiRoute.scala [hunk 3]
```diff
@@ -66,26 +66,30 @@ case class TransactionsApiRoute(
...
+          val tx = either.fold(identity, _._1)
+          complete(txToExtendedJson(tx) ++ enrich(tx) ++ applicationStatusJsField(isBlockV5(h), succeeded) + ("height" -> JsNumber(h)))
         case None => complete(ApiError.TransactionDoesNotExist)
       }
     }
   }
 
+  private def enrich(t: Transaction) = {
+    t match {
+      case t: smart.InvokeScriptTransaction => Json.obj("сontinuationTransactionIds" -> commonApi.continuations(t.id()).map(_.toString))
```
Comment (xrtm000) [inline node/src/main/scala/com/wavesplatform/api/http/TransactionsApiRoute.scala:78]: It is unnecessary field for old InvokeScript transactions that don't support continuations. Introduce InvokeScript transaction V3 to distinguish.

**Review by xrtm000 — CHANGES_REQUESTED** [review]

**Review by xrtm000 — CHANGES_REQUESTED** [review]
Please, add also **feeAssetId**, **version** and **height** fields

File: node/src/main/scala/com/wavesplatform/transaction/serialization/impl/ContinuationTxSerializer.scala [hunk 4]
```diff
@@ -7,6 +7,7 @@ object ContinuationTxSerializer {
   def toJson(tx: ContinuationTransaction): JsObject = {
     import tx._
     Json.obj(
+      "version"                   -> 1,
```
Comment (xrtm000) [inline node/src/main/scala/com/wavesplatform/transaction/serialization/impl/ContinuationTxSerializer.scala:10]: Maybe add this field directly to ContinuationTransaction class?

**Comment by xrtm000** [comment]
And test new fields, please

**Review by xrtm000 — CHANGES_REQUESTED** [review]
Please, add height and test new fields

**Comment by xrtm000** [comment]
Test **dApp** and **call** fields

**Review by xrtm000 — APPROVED** [review]

**Review by mrkraft — APPROVED** [review]

**Pull request merged**